packages/cosmic-swingset/lib/ag-solo/vats/solo-config.json

```diff
@@ -0,0 +1,49 @@
+{
+  "bootstrap": "bootstrap",
+  "bundles": {
+    "zcf": {
+      "sourcePath": "../../../../zoe/src/contractFacet.js"
```

Comment:
I think anything going across package boundaries should use the npm import and not a local sourcePath.

Reply:
If there were a way to do that, it would make me happier.  However, since this code doesn't enter our world via the import mechanism that's not an option (at least, not right now -- one notion on my todo list is to wire up the config file processor's path resolution mechanism to use npm's; this may have been what Brian was talking about the other day in reference to using `require.resolve` in a related context.  I will investigate.)

Reply:
Ah, that's right. That makes sense.

Reply:
Out of curiosity, I just tried `require.resolve('@agoric/zoe/src/contractFacet')` and it returns the absolute path to `contractFact.js`, so this may be easier than I feared.  I still don't care for the link referencing the `src` directory directly, but it at least gets rid of the big dangerous `../../../../` thing.

Reply:
I don't want to depend upon the internal layout of foreign packages: at some point zoe will move that file elsewhere and then this will break. Intead, I'd like to have zoe export a function like `zcfBundlePathname()`, which internally uses `require.resolve('./path/to/my/zcf.js')`. Then cosmic-swingset can import that function, call it, and add the result into `config.bundles.zcf`.

I don't know how to square that against using a JSON config file for everything, but I think a reasonable compromise might be to load the JSON first, and then run the function to add the additional property afterwards. Not as elegant, so I guess it's a judgement call against the maintenance hazard of the deep-import.

A slightly different approach might be if Zoe were to explicitly expose the `zcf` file, in its `package.json`, with the "`exports:`" directive. Currently Node implicitly exports every file in the package one-to-one, but Endo will only let you import things that were explicitly exported. With the `exports:` directive you can control the mapping, so when someone outside says `import '@zoe/foo'`, they really get `zoe/src/bar` or whatever. That would make it more obvious (to future Zoe authors) that they'd committed to making that file available via that specific deep-import, and would allow internal rearrangement to happen without breaking the external importers. However it still requires changing the cosmic-swingset code to run a function and modify the config object, rather than getting everything from a JSON file.

Reply:
I'd prefer to have (not this exact code):
```js
function resolveSpecifier(specifier) {
  if (specifier.match(/^\.\.?(\/|$)/)) {
   // relative specifier: resolve to file?
  } else if (specifier.startsWith('/')) {
   // absolute specifier: resolve to file?
  } else {
   // base specifier
   return require.resolve(specifier);
  }
}
```
and then to run that function on everything that is a "path" in the JSON file.

This logic would be fully general, rather than coding specially for Zcf.

Reply:
> I don't want to depend upon the internal layout of foreign packages

We're in a monorepo.  This is not a foreign package, and CI will catch us if it breaks.  Also, conceivably, people will want to deep-link to their own packages for their own vat bundles, and it makes sense to support this.

Reply:
That sounds like an improvement. I'm not entirely sure about the `require.resolve(specifier)` line, that will be interpreted relative to the bit inside SwingSet where `resolveSpecifier` lives, so `./kernel/kernel.js` or `./../../controller.js` depends upon kernel file layout that should really be private).

The combination of `resolveSpecifier` and Zoe's package.json having an `export:` directive could allow cosmic-swingset's static `package.json` to cite `@agoric/zoe/zcf-facet`, but that would only work "properly" if swingset itself had a dependency on Zoe, because `'@agoric/zoe'` anything is going to be looked up relative to *swingset*'s file, not cosmic-swingset's. It would probably work in the monorepo, because the top-level `node_modules` includes zoe. But in Endo's more-explicit-everything world, it might not (if swingset wants to import something from zoe, swingset needs to have a dependency on zoe).  That will be the case in any solution unless the the `require.resolve` appears within cosmic-swingset.

Reply:
> I'm not entirely sure about the `require.resolve(specifier)` line

Okay, you've prodded me into updating my pseudocode into [actually working code](https://github.com/Agoric/agoric-sdk/pull/1439#discussion_r469480426).  No need for extraneous dependencies in SwingSet.

Reply:
Everybody's getting ahead of the game.  I've already got something waiting to go into the next PR in this sequence.

Reply:
> I've already got something waiting to go into the next PR in this sequence.

Given that you are addressing it, I will approve this PR.